virttest/utils_kernel_module.py

```diff
@@ -0,0 +1,188 @@
+                do_not_load = True
+            elif not self._was_loaded:
+                logging.debug("Module %s isn't loaded. Use force=True to force"
+                              " loading.", self._module_name)
+                do_not_load = True
+            if do_not_load:
+                return
+
+        cmd = ""
+        if self._was_loaded:
+            cmd += 'rmmod %s; ' % self._module_name
+        cmd += 'modprobe %s %s' % (self._module_name, params)
+        cmd = cmd.strip()
+        logging.debug("Loading module: %s", cmd)
+        status, output = process.getstatusoutput(cmd, shell=True,
+                                                 ignore_status=True)
+        if status:
+            raise exceptions.TestError("Couldn't load module %s: %s" % (
+                self._module_name, output
+            ))
+        self._loaded_config = params
+
+    def restore(self):
+        """
+        Restore previous module state.
+
+        The state will only be restored if the original state
+        was altered.
+
+        +-------------------+-+-+-+-+
+        |**precondition**           |
+        +-------------------+-+-+-+-+
+        |module loaded      |Y|Y|N|N|
+        +-------------------+-+-+-+-+
+        |loaded with params |Y|N|Y|N|
+        +-------------------+-+-+-+-+
+        |**result**                 |
+        +-------------------+-+-+-+-+
+        |restore            |Y|N|Y|N|
+        +-------------------+-+-+-+-+
+        """
+
+        if self._loaded_config is not None:
+            if not self._was_loaded:
+                cmd = 'rmmod %s' % self._module_name
+            else:
+                cmd = 'rmmod %s; modprobe %s %s' % (self._module_name,
+                                                    self._module_name,
+                                                    self._config_backup)
+            cmd = cmd.strip()
+            logging.debug("Restoring module state: %s", cmd)
+            status = process.system(cmd, shell=True,
+                                    ignore_status=True)
+            if status:
+                raise exceptions.TestError("Couldn't restore module %s: %s" % (
+                    self._module_name, self._config_backup))
+
+    def _backup_config(self):
+        """
+        Check if self.module_name is loaded and read config
+
+        """
+        config = self._get_serialized_config()
+        if config is not None:
+            self._config_backup = config
+            self._was_loaded = True
+        else:
+            self._was_loaded = False
+        logging.debug("Backed up %s module state (was_loaded, params)"
+                      "=(%s, %s)", self._module_name, self._was_loaded,
+                      self._config_backup)
+
+    def get_was_loaded(self):
+        """ Read-only property """
+
+        return self._was_loaded
+
+    def get_config_backup(self):
+        """ Read-only property """
+
+        return self._config_backup
```

Comment:
note: this reminds me java constructs, in python it's more common to actually use `@property` without `get_` prefix